libminifi/src/EventDrivenSchedulingAgent.cpp

```diff
@@ -29,6 +29,13 @@ namespace apache {
 namespace nifi {
 namespace minifi {
 
+void EventDrivenSchedulingAgent::schedule(std::shared_ptr<core::Processor> processor) {
+  if (!processor->hasIncomingConnections()) {
+    throw Exception(PROCESS_SCHEDULE_EXCEPTION, "EventDrivenSchedulingAgent cannot schedule processor without incoming connection!");
```

Comment:
I suggest logging the processor UUID and possibly its type as well. I think it would be helpful to users with nontrivial flows who happened to incorrectly configure minifi c++.

Reply:
Exactly that happens where the exception is handled (ProcessGroup.cpp):

```
    catch (const std::exception &e) {
      logger_->log_error("Failed to start processor %s (%s): %s", processor->getUUIDStr(), processor->getName(), e.what());
      failed_processors.insert(processor);
    }
```